client/components/split-button/index.jsx

```diff
@@ -91,15 +92,19 @@ class SplitButton extends PureComponent {
 			'has-icon-text': label && icon,
 		} );
 
+		const isEmptyOnClick = this.props.onClick === noop;
+		const onClick = isEmptyOnClick ? undefined : this.handleMainClick;
+
 		return (
 			<span className={ classes }>
 				<Button
 					compact={ compact }
 					primary={ primary }
 					scary={ scary }
-					onClick={ this.handleMainClick }
 					disabled={ disabled || disableMain }
 					className="split-button__main"
+					onClick={ onClick }
+					href={ this.props.href }
```

Comment:
I see that in other places in Calypso, we are using the `onClick` callback to handle actions related to the navigation, e.g.:

https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/activity/activity-log-tasklist/index.jsx#L349

Would it make sense to follow the same way instead of adding support for the `href` to the component? That way, it may be easier to add more logic around that e.g., we can easily add tracking to that action.

Reply:
Yes, and I think it adds an extra complexity that is not necessary.
The `SplitButton` uses the Automattic Button component under the hood, which already supports the `href` prop, and it renders a link instead of a button which is more semantic, and it also tracks all the navigations automatically.